File: frontend/src/apis/video.ts
import {request} from "../utils/requests";
import configs from "../configs";

export async function getVideos(force: boolean = false) {
    const response = await request.request({
        url: '/video/',
        method: 'get',
        params: {force}
    })
    return response.data.data
}

export async function getVideoDetail(path: string) {
    const response = await request.request({
        url: '/video/detail',
        method: 'get',
        params: {path}
    })
    return response.data.data
}

export async function parseVideoNum(path: string) {
    const response = await request.request({
        url: '/video/parse',
        method: 'get',
        params: {path}
    })
    return response.data.data
}

export function getVideoCover(url: string) {
    return configs.BASE_API + '/common/cover?url=' + encodeURIComponent(url)
}

export function scrapeVideo(num: string) {
    return request.request({
        url: '/video/scrape',
        method: 'get',
        params: {num}
    })
}

export function saveVideo(data: any, mode?: string, transMode?: string) {
    return request.request({
        url: '/video/',
        method: 'post',
        params: {mode, trans_mode: transMode},
        data: data
    })
}

export function deleteVideo(path?: string) {
    return request.request({
        url: '/video/',
        method: 'delete',
        params: {path},
    })
}

File: app/api/video.py
from typing import Optional

from fastapi import APIRouter, Depends

from app.schema import VideoDetail
from app.schema.r import R
from app.service.video import get_video_service

router = APIRouter()


@router.get('/')
def get_videos(force: Optional[bool] = False, service=Depends(get_video_service)):
    if force:
        video = service.get_videos_force()
    else:
        video = service.get_videos()
    return R.list(video)


@router.get('/detail')
def get_video(path: str, service=Depends(get_video_service)):
    video = service.get_video(path)
    return R.ok(video)


@router.get('/parse')
def parse_video(path: str, service=Depends(get_video_service)):
    video = service.parse_video(path)
    return R.ok(video)


@router.get('/scrape')
def scrape_video(num: str, service=Depends(get_video_service)):
    video = service.scrape_video(num)
    return R.ok(video)


@router.post('/')
def save_video(video: VideoDetail,
               mode: Optional[str] = None,
               trans_mode: Optional[str] = None,
               service=Depends(get_video_service)):
    service.save_video(video, mode, trans_mode)
    return R.ok()


@router.delete('/')
def delete_video(path: str, service=Depends(get_video_service)):
    service.delete_video(path)
    return R.ok()
